PMKVObserverTests/PMKVObserverTests.m

```diff
@@ -239,4 +239,67 @@ - (void)testInitialCancel {
     XCTAssertNil(weakToken);
 }
 
+- (void)testCancelOnBackgroundThread {
+    __block BOOL fired = NO;
+    dispatch_queue_t queue = dispatch_queue_create("com.postmates.PMKVObserver.queue", DISPATCH_QUEUE_SERIAL);
+    NSString* value1 = @"value1";
+    NSString* value2 = @"value2";
+
+    PMKVObserver *token = [PMKVObserver observeObject:self.helper keyPath:@"str" options:NSKeyValueObservingOptionNew block:^(id  _Nonnull object, NSDictionary<NSString *,id> * _Nullable change, PMKVObserver * _Nonnull kvo) {
+        XCTAssertEqualObjects([object str], value1);
+        fired = YES;
+    }];
+    dispatch_sync(queue, ^{
+        self.helper.str = value1;
+        [token cancel];
```

Comment:
This isn't actually doing anything on a background thread at all. `dispatch_sync()` always executes the block on the current thread unless the queue being disaptched to is the main queue (and the process is using `NSRunLoop` or `CFRunLoop`; a process using `dispatch_main` doesn't tie the main queue to thread 0).